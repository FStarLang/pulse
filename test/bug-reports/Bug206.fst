module Bug206

#lang-pulse
open Pulse

fn test (x:nat { x > 0 })
requires emp
ensures emp
{
  ()
}

[@@expect_failure [19]]
fn use_test (x:nat)
requires emp
ensures emp
{
  test x
}
