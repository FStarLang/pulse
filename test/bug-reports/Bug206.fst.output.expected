>> Got issues: [
* Error 19 at Bug206.fst(18,7-18,8):
  - Ill-typed term: x
  - Expected a term of type x: Prims.nat{x > 0}
  - Assertion failed
  - The SMT solver could not prove the query. Use --query_stats for more
    details.
  - Also see: Bug206.fst(18,2-18,6)
  - Other related locations: Bug206.fst(6,20-6,20)
  - See also Bug206.fst(18,2-18,8)

>>]
